golang/trace/zipin_tracereporter_test.go

```diff
@@ -163,30 +230,21 @@ func withSetup(t *testing.T, f func(ctx thrift.Context, args testArgs)) {
 	args.s.AssertExpectations(t)
 }
 
-func setupServer(h *mocks.TChanTCollector) (*tchannel.Channel, net.Listener, error) {
-	tchan, err := tchannel.NewChannel(tcollectorServiceName, &tchannel.ChannelOptions{
-		Logger: tchannel.SimpleLogger,
+func setupServer(h *mocks.TChanTCollector) (*tchannel.Channel, error) {
+	tchan, err := testutils.NewServer(&testutils.ChannelOpts{
+		ServiceName: tcollectorServiceName,
 	})
 	if err != nil {
-		return nil, nil, err
-	}
-
-	listener, err := net.Listen("tcp", ":0")
-	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	server := thrift.NewServer(tchan)
 	server.Register(gen.NewTChanTCollectorServer(h))
```

Comment:
`testutils.NewServer` just sets up a listening tchannel (no thrift setup or anything). It is mostly used to get rid of some boilerplate and has disabled logging by default, but you can enable it by adding `--connectionLog`. That way tests are quiet and I can enable logging when debugging tests.